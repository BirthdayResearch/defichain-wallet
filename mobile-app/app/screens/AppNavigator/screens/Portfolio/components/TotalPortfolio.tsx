import { View } from "@components";
import { TextSkeletonLoader } from "@components/TextSkeletonLoader";
import { ThemedIcon, ThemedTextV2, ThemedViewV2 } from "@components/themed";
import { RootState } from "@store";
import { tailwind } from "@tailwind";
import { translate } from "@translations";
import BigNumber from "bignumber.js";
import { NumericFormat as NumberFormat } from "react-number-format";
import { useSelector } from "react-redux";
import { Platform, TouchableOpacity } from "react-native";
import { useEffect, useState } from "react";
import { TextSkeletonLoaderV2 } from "@components/TextSkeletonLoaderV2";
import { useDomainContext, DomainType } from "@contexts/DomainContext";
import {
  getPrecisedCurrencyValue,
  getPrecisedTokenValue,
} from "../../Auctions/helpers/precision-token-value";
import { BalanceText } from "./BalanceText";
import { BalanceTextV2 } from "./BalanceTextV2";

export enum PortfolioButtonGroupTabKey {
  USDT = "USDT",
  DFI = "DFI",
  BTC = "BTC",
  DUSD = "DUSD",
  USDC = "USDC",
  EUROC = "EUROC",
}

interface TotalPortfolioProps {
  totalAvailableValue: BigNumber;
  totalLockedValue: BigNumber;
  totalLoansValue: BigNumber;
  portfolioButtonGroup: PortfolioButtonGroup[];
  denominationCurrency: string;
  setDenominationCurrency: (key: PortfolioButtonGroupTabKey) => void;
}

interface PortfolioButtonGroup {
  id: PortfolioButtonGroupTabKey;
  label: string;
  handleOnPress: () => void;
}

export function TotalPortfolio(props: TotalPortfolioProps): JSX.Element {
  const { hasFetchedToken } = useSelector((state: RootState) => state.wallet);
  const { domain } = useDomainContext();
  const isEvmDomain = domain === DomainType.EVM;
  const { hasFetchedVaultsData } = useSelector(
    (state: RootState) => state.loans,
  );
  const [isExpanded, setIsExpanded] = useState<boolean>(false);
  const denominationCurrency = props.denominationCurrency; // for 'BTC' or 'DFI' denomination
  const totalPortfolioValue = isEvmDomain
    ? new BigNumber(props.totalAvailableValue)
    : BigNumber.max(
        0,
        new BigNumber(props.totalAvailableValue)
          .plus(props.totalLockedValue)
          .minus(props.totalLoansValue),
      );
  const [activeButtonGroup, setActiveButtonGroup] =
    useState<PortfolioButtonGroup>();
  const onCurrencySwitch = (): void => {
    const activeIndex = props.portfolioButtonGroup.findIndex(
      (tab) => tab.id === props.denominationCurrency,
    );
    let nextIndex = activeIndex + 1;
    if (activeIndex === props.portfolioButtonGroup.length - 1) {
      nextIndex = 0;
    }
    props.setDenominationCurrency(props.portfolioButtonGroup[nextIndex].id);
  };

  useEffect(() => {
    setActiveButtonGroup(
      props.portfolioButtonGroup.find(
        (button) => button.id === props.denominationCurrency,
      ),
    );
  }, [props.denominationCurrency]);

  return (
    <ThemedViewV2
      light={tailwind("bg-mono-light-v2-00")}
      dark={tailwind("bg-mono-dark-v2-00")}
      style={tailwind("px-5 py-5 rounded-b-xl-v2")}
      testID="total_portfolio_card"
    >
      {hasFetchedToken && hasFetchedVaultsData ? (
        <View style={tailwind("flex flex-row items-center justify-between")}>
          <TouchableOpacity
            activeOpacity={0.7}
            onPress={onCurrencySwitch}
            style={tailwind("flex flex-row items-center w-10/12")}
            testID="portfolio_currency_switcher"
          >
            <NumberFormat
              displayType="text"
              prefix={
                denominationCurrency === PortfolioButtonGroupTabKey.USDT
                  ? "$"
                  : undefined
              }
              renderText={(value) => (
                <BalanceTextV2
                  style={[
                    tailwind(
                      "font-semibold-v2 mr-2 flex flex-row items-center",
                    ),
                    { fontSize: 28, lineHeight: 36 },
                  ]}
                  testID="total_usd_amount"
                  value={value}
                >
                  {activeButtonGroup !== undefined && (
                    <CurrencySwitcher currency={activeButtonGroup.label} />
                  )}
                </BalanceTextV2>
              )}
              thousandSeparator
              value={
                denominationCurrency === PortfolioButtonGroupTabKey.USDT
                  ? getPrecisedCurrencyValue(totalPortfolioValue)
                  : getPrecisedTokenValue(totalPortfolioValue)
              }
            />
          </TouchableOpacity>
          {!isEvmDomain && (
            <TouchableOpacity
              activeOpacity={0.7}
              onPress={() => setIsExpanded(!isExpanded)}
              style={tailwind("")}
              testID="toggle_portfolio"
            >
              <ThemedIcon
                light={tailwind("text-mono-light-v2-900")}
                dark={tailwind("text-mono-dark-v2-900")}
                iconType="Feather"
                name={!isExpanded ? "chevron-down" : "chevron-up"}
                size={30}
              />
            </TouchableOpacity>
          )}
        </View>
      ) : (
        <View style={tailwind("mt-1")}>
          <TextSkeletonLoaderV2
            viewBoxWidth="260"
            viewBoxHeight="32"
            textWidth="180"
            textHeight="20"
            textVerticalOffset="4"
            iContentLoaderProps={{
              height: "32",
              testID: "total_portfolio_skeleton_loader",
            }}
          />
        </View>
      )}
      {isExpanded && !isEvmDomain && (
        <ThemedViewV2
          style={tailwind("mt-5 border-t-0.5")}
          light={tailwind("border-mono-light-v2-300")}
          dark={tailwind("border-mono-dark-v2-300")}
        >
          <View style={tailwind("mt-5")}>
            <USDValueRow
              testId="total_available_usd_amount"
              isLoading={!hasFetchedToken}
              label={translate("screens/PortfolioScreen", "available")}
              value={props.totalAvailableValue}
              isAddition
              denominationCurrency={denominationCurrency}
            />
          </View>
          <USDValueRow
            testId="total_locked_usd_amount"
            isLoading={!hasFetchedVaultsData}
            label={translate("screens/PortfolioScreen", "locked in vault(s)")}
            value={props.totalLockedValue}
            isAddition
            denominationCurrency={denominationCurrency}
          />
          {props.totalLoansValue.gt(0) && (
            <USDValueRow
              testId="outstanding_loans_amount"
              isLoading={!hasFetchedVaultsData}
              label={translate("screens/PortfolioScreen", "loans")}
              value={props.totalLoansValue}
              isAddition={false}
              denominationCurrency={denominationCurrency}
            />
          )}
        </ThemedViewV2>
      )}
    </ThemedViewV2>
  );
}

function USDValueRow(props: {
  isLoading: boolean;
  testId: string;
  value: BigNumber;
  label: string;
  isAddition: boolean;
  denominationCurrency?: string;
}): JSX.Element {
  if (props.isLoading) {
    return (
      <View style={tailwind("mt-1")}>
        <TextSkeletonLoader
          iContentLoaderProps={{
            height: "14",
            testID: `${props.testId}_skeleton_loader`,
          }}
          viewBoxWidth="260"
          textWidth="120"
        />
      </View>
    );
  }
  return (
    <View style={tailwind("flex flex-row justify-start items-center w-full")}>
      <ThemedTextV2
        light={tailwind(props.isAddition ? "text-green-v2" : "text-red-v2")}
        dark={tailwind(props.isAddition ? "text-green-v2" : "text-red-v2")}
        style={tailwind("mr-1 w-2 text-sm font-normal-v2")}
      >
        {props.isAddition ? "+" : "-"}
      </ThemedTextV2>
      <NumberFormat
        displayType="text"
        // TODO: modify condition when API is ready for denomination currency change for other pages other than BalanceScreen page
        prefix={
          props.denominationCurrency === undefined ||
          props.denominationCurrency === PortfolioButtonGroupTabKey.USDT
            ? "$"
            : undefined
        }
        suffix={
          props.denominationCurrency !== undefined &&
          props.denominationCurrency !== PortfolioButtonGroupTabKey.USDT
            ? ` ${props.denominationCurrency}`
            : undefined
        }
        renderText={(value) => (
          <BalanceText
            light={tailwind(props.isAddition ? "text-green-v2" : "text-red-v2")}
            dark={tailwind(props.isAddition ? "text-green-v2" : "text-red-v2")}
            style={tailwind("flex-wrap text-sm font-normal-v2 ")}
            testID={props.testId}
            value={value}
          />
        )}
        thousandSeparator
        value={
          props.denominationCurrency === PortfolioButtonGroupTabKey.USDT
            ? getPrecisedCurrencyValue(props.value)
            : getPrecisedTokenValue(props.value)
        }
      />
      <ThemedTextV2 style={tailwind("text-sm font-normal-v2 ml-1")}>
        {props.label}
      </ThemedTextV2>
    </View>
  );
}

function CurrencySwitcher({ currency }: { currency: string }): JSX.Element {
  return (
    <ThemedViewV2
      style={tailwind("py-1 px-2 rounded-lg border-0.5 self-center", {
        "-mb-1.5": Platform.OS === "android",
      })}
      light={tailwind("border-mono-light-v2-900")}
      dark={tailwind("border-mono-dark-v2-900")}
    >
      <ThemedTextV2
        style={tailwind("text-xs font-normal-v2")}
        testID="portfolio_active_currency"
      >
        {currency}
      </ThemedTextV2>
    </ThemedViewV2>
  );
}
