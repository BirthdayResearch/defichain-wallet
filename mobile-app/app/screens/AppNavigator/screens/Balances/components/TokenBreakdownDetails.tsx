import { View } from '@components'
import { TextSkeletonLoader } from '@components/TextSkeletonLoader'
import { ThemedProps, ThemedText } from '@components/themed'
import { WalletToken } from '@store/wallet'
import { tailwind } from '@tailwind'
import { translate } from '@translations'
import BigNumber from 'bignumber.js'
import { StyleProp, TextProps, ViewProps } from 'react-native'
import NumberFormat from 'react-number-format'
import { BalanceText } from './BalanceText'
import { PortfolioButtonGroupTabKey } from './TotalPortfolio'

interface TokenBreakdownDetailProps {
  hasFetchedToken: boolean
  lockedAmount: BigNumber
  lockedValue: BigNumber
  availableAmount: BigNumber
  availableValue: BigNumber
  dfiUtxo?: WalletToken
  dfiToken?: WalletToken
  testID: string
  denominationCurrency: string
}

export function TokenBreakdownDetails (props: TokenBreakdownDetailProps): JSX.Element {
  return (
    <>
      <TokenBreakdownDetailsRow
        testID={`${props.testID}_locked`}
        amount={props.lockedAmount.toFixed(8)}
        label='Locked in vault(s)'
        hasFetchedToken={props.hasFetchedToken}
        labelTextStyle={tailwind('font-medium')}
        valueThemeProps={{
          light: tailwind('text-black'),
          dark: tailwind('text-white')
        }}
      />
      <TokenBreakdownDetailsRow
        testID={`${props.testID}_locked_value`}
        amount={props.lockedValue.toFixed(8)}
        label=''
        hasFetchedToken={props.hasFetchedToken}
        valueThemeProps={{
          light: tailwind('text-gray-500'),
          dark: tailwind('text-gray-400')
        }}
        containerStyle={tailwind('mb-2')}
        prefix={props.denominationCurrency === PortfolioButtonGroupTabKey.USDT ? '≈ $' : undefined}
        suffix={props.denominationCurrency !== PortfolioButtonGroupTabKey.USDT ? ` ${props.denominationCurrency}` : undefined}
      />
      <TokenBreakdownDetailsRow
        testID={`${props.testID}_available`}
        amount={props.availableAmount.toFixed(8)}
        label='Available'
        hasFetchedToken={props.hasFetchedToken}
        labelTextStyle={tailwind('font-medium')}
        valueThemeProps={{
          light: tailwind('text-black'),
          dark: tailwind('text-white')
        }}
      />
      <TokenBreakdownDetailsRow
        testID={`${props.testID}_available_value`}
        amount={props.availableValue.toFixed(8)}
        label=''
        hasFetchedToken={props.hasFetchedToken}
        valueThemeProps={{
          light: tailwind('text-gray-500'),
          dark: tailwind('text-gray-400')
        }}
        prefix={props.denominationCurrency === PortfolioButtonGroupTabKey.USDT ? '≈ $' : undefined}
        suffix={props.denominationCurrency !== PortfolioButtonGroupTabKey.USDT ? ` ${props.denominationCurrency}` : undefined}
      />
      {props.dfiUtxo !== undefined && props.dfiToken !== undefined &&
        (
          <View style={tailwind('mt-4')}>
            <TokenBreakdownDetailsRow testID='dfi_utxo' amount={props.dfiUtxo.amount} label='as UTXO' hasFetchedToken={props.hasFetchedToken} containerStyle={tailwind('mb-1')} />
            <TokenBreakdownDetailsRow testID='dfi_token' amount={props.dfiToken.amount} label='as Token' hasFetchedToken={props.hasFetchedToken} />
          </View>
        )}
    </>
  )
}

interface TokenBreakdownDetailsRowProps {
  amount: string
  label: string
  testID: string
  hasFetchedToken: boolean
  labelTextStyle?: StyleProp<TextProps>
  valueTextStyle?: StyleProp<TextProps>
  valueThemeProps?: ThemedProps
  containerStyle?: StyleProp<ViewProps>
  prefix?: string
  suffix?: string
}
function TokenBreakdownDetailsRow ({
  amount,
  label,
  testID,
  hasFetchedToken,
  labelTextStyle,
  valueTextStyle,
  valueThemeProps = {
    light: tailwind('text-gray-500'),
    dark: tailwind('text-gray-400')
  },
  containerStyle,
  prefix,
  suffix
}: TokenBreakdownDetailsRowProps): JSX.Element {
  return (
    <View style={[tailwind('flex-row flex-1 items-center'), containerStyle]}>
      <ThemedText
        light={tailwind('text-gray-500')}
        dark={tailwind('text-gray-400')}
        style={[tailwind('pr-14 text-xs'), labelTextStyle]}
        testID={`${testID}_label`}
      >
        {translate('components/DFIBalanceCard', label)}
      </ThemedText>
      <View style={tailwind('flex-row flex-1 justify-end')}>
        {
          hasFetchedToken
            ? (
              <NumberFormat
                value={amount}
                thousandSeparator
                decimalScale={8}
                fixedDecimalScale
                displayType='text'
                prefix={prefix}
                suffix={suffix}
                renderText={value =>
                  <BalanceText
                    light={valueThemeProps.light}
                    dark={valueThemeProps.dark}
                    style={[tailwind('text-xs'), valueTextStyle]}
                    testID={`${testID}_amount`}
                    value={value}
                  />}
              />
            )
            : (
              <TextSkeletonLoader
                iContentLoaderProps={{
                  width: '150',
                  height: '14',
                  testID: 'dfi_breakdown_row_skeleton_loader'
                }}
                textHorizontalOffset='30'
                textWidth='120'
              />
            )
        }
      </View>
    </View>
  )
}
