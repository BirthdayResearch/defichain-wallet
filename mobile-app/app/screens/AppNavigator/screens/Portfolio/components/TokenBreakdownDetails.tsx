import { View } from "@components";
import { TextSkeletonLoader } from "@components/TextSkeletonLoader";
import { ThemedProps, ThemedText } from "@components/themed";
import { WalletToken } from "@waveshq/walletkit-ui/dist/store";
import { tailwind } from "@tailwind";
import { translate } from "@translations";
import BigNumber from "bignumber.js";
import { StyleProp, TextProps, ViewStyle } from "react-native";
import { NumericFormat as NumberFormat } from "react-number-format";
import { getPrecisedTokenValue } from "../../Auctions/helpers/precision-token-value";
import { BalanceText } from "./BalanceText";
import { PortfolioButtonGroupTabKey } from "./TotalPortfolio";

interface TokenBreakdownDetailProps {
  hasFetchedToken: boolean;
  lockedAmount: BigNumber;
  lockedValue: BigNumber;
  availableAmount: BigNumber;
  availableValue: BigNumber;
  dfiUtxo?: WalletToken;
  dfiToken?: WalletToken;
  testID: string;
  denominationCurrency: string;
}

export function TokenBreakdownDetails(
  props: TokenBreakdownDetailProps
): JSX.Element {
  return (
    <>
      <TokenBreakdownDetailsRow
        testID={`${props.testID}_locked`}
        amount={props.lockedAmount.toFixed(8)}
        label="Locked"
        hasFetchedToken={props.hasFetchedToken}
        labelTextStyle={tailwind("font-medium")}
        valueThemeProps={{
          light: tailwind("text-black"),
          dark: tailwind("text-white"),
        }}
      />
      <TokenBreakdownDetailsRow
        testID={`${props.testID}_locked_value`}
        amount={getPrecisedTokenValue(props.lockedValue)}
        label=""
        hasFetchedToken={props.hasFetchedToken}
        valueThemeProps={{
          light: tailwind("text-gray-500"),
          dark: tailwind("text-gray-400"),
        }}
        containerStyle={tailwind("mb-2")}
        prefix={
          props.denominationCurrency === PortfolioButtonGroupTabKey.USDT
            ? "≈ $"
            : undefined
        }
        suffix={
          props.denominationCurrency !== PortfolioButtonGroupTabKey.USDT
            ? ` ${props.denominationCurrency}`
            : undefined
        }
      />
      <TokenBreakdownDetailsRow
        testID={`${props.testID}_available`}
        amount={props.availableAmount.toFixed(8)}
        label="Available"
        hasFetchedToken={props.hasFetchedToken}
        labelTextStyle={tailwind("font-medium")}
        valueThemeProps={{
          light: tailwind("text-black"),
          dark: tailwind("text-white"),
        }}
      />
      <TokenBreakdownDetailsRow
        testID={`${props.testID}_available_value`}
        amount={getPrecisedTokenValue(props.availableValue)}
        label=""
        hasFetchedToken={props.hasFetchedToken}
        valueThemeProps={{
          light: tailwind("text-gray-500"),
          dark: tailwind("text-gray-400"),
        }}
        prefix={
          props.denominationCurrency === PortfolioButtonGroupTabKey.USDT
            ? "≈ $"
            : undefined
        }
        suffix={
          props.denominationCurrency !== PortfolioButtonGroupTabKey.USDT
            ? ` ${props.denominationCurrency}`
            : undefined
        }
      />
      {props.dfiUtxo !== undefined && props.dfiToken !== undefined && (
        <View style={tailwind("mt-4")}>
          <TokenBreakdownDetailsRow
            testID="dfi_utxo"
            amount={new BigNumber(props.dfiUtxo.amount).toFixed(8)}
            label="as UTXO"
            hasFetchedToken={props.hasFetchedToken}
            containerStyle={tailwind("mb-1")}
          />
          <TokenBreakdownDetailsRow
            testID="dfi_token"
            amount={new BigNumber(props.dfiToken.amount).toFixed(8)}
            label="as Token"
            hasFetchedToken={props.hasFetchedToken}
          />
        </View>
      )}
    </>
  );
}

interface TokenBreakdownDetailsRowProps {
  amount: string;
  label: string;
  testID: string;
  hasFetchedToken: boolean;
  labelTextStyle?: StyleProp<TextProps>;
  valueTextStyle?: StyleProp<TextProps>;
  valueThemeProps?: ThemedProps;
  containerStyle?: StyleProp<ViewStyle>;
  prefix?: string;
  suffix?: string;
}
function TokenBreakdownDetailsRow({
  amount,
  label,
  testID,
  hasFetchedToken,
  labelTextStyle,
  valueTextStyle,
  valueThemeProps = {
    light: tailwind("text-gray-900"),
    dark: tailwind("text-gray-50"),
  },
  containerStyle,
  prefix,
  suffix,
}: TokenBreakdownDetailsRowProps): JSX.Element {
  return (
    <View style={[tailwind("flex-row flex-1 items-center"), containerStyle]}>
      <ThemedText
        light={tailwind("text-gray-500")}
        dark={tailwind("text-gray-400")}
        style={[tailwind("pr-14 text-xs"), labelTextStyle]}
        testID={`${testID}_label`}
      >
        {translate("components/DFIBalanceCard", label)}
      </ThemedText>
      <View style={tailwind("flex-row flex-1 justify-end")}>
        {hasFetchedToken ? (
          <NumberFormat
            value={amount}
            thousandSeparator
            displayType="text"
            prefix={prefix}
            suffix={suffix}
            renderText={(value) => (
              <BalanceText
                light={valueThemeProps.light}
                dark={valueThemeProps.dark}
                style={[tailwind("text-xs"), valueTextStyle]}
                testID={`${testID}_amount`}
                value={value}
              />
            )}
          />
        ) : (
          <TextSkeletonLoader
            iContentLoaderProps={{
              width: "150",
              height: "14",
              testID: "dfi_breakdown_row_skeleton_loader",
            }}
            textHorizontalOffset="30"
            textWidth="120"
          />
        )}
      </View>
    </View>
  );
}
