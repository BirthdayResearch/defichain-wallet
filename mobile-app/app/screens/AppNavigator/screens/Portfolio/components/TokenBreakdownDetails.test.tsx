import { render } from "@testing-library/react-native";
import BigNumber from "bignumber.js";
import { TokenBreakdownDetails } from "./TokenBreakdownDetails";
import { PortfolioButtonGroupTabKey } from "./TotalPortfolio";

jest.mock("../../../../../contexts/DisplayBalancesContext");

describe("Token Breakdown Details", () => {
  it("should match snapshot", async () => {
    const component = (
      <TokenBreakdownDetails
        hasFetchedToken
        lockedAmount={new BigNumber("1")}
        lockedValue={new BigNumber("123")}
        availableAmount={new BigNumber("99")}
        availableValue={new BigNumber("999999.12345678")}
        testID="foo"
        denominationCurrency={PortfolioButtonGroupTabKey.USDT}
      />
    );
    const rendered = render(component);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
