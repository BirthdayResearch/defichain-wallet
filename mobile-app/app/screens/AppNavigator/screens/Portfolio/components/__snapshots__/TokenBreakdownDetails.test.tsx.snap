// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Token Breakdown Details should match snapshot 1`] = `
[
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexBasis": "0%",
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          [
            [
              {
                "fontSize": 12,
                "lineHeight": 16,
                "paddingRight": 56,
              },
              {
                "fontFamily": "MediumFont",
                "fontWeight": "500",
              },
            ],
            {
              "color": "rgba(115, 115, 115, 1)",
            },
          ],
        ]
      }
      testID="foo_locked_label"
    >
      Locked
    </Text>
    <View
      style={
        {
          "flexBasis": "0%",
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            [
              [
                {
                  "fontSize": 12,
                  "lineHeight": 16,
                },
                undefined,
              ],
              {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
          ]
        }
        testID="foo_locked_amount"
      >
        1.00000000
      </Text>
    </View>
  </View>,
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexBasis": "0%",
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "marginBottom": 8,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          [
            [
              {
                "fontSize": 12,
                "lineHeight": 16,
                "paddingRight": 56,
              },
              undefined,
            ],
            {
              "color": "rgba(115, 115, 115, 1)",
            },
          ],
        ]
      }
      testID="foo_locked_value_label"
    />
    <View
      style={
        {
          "flexBasis": "0%",
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            [
              [
                {
                  "fontSize": 12,
                  "lineHeight": 16,
                },
                undefined,
              ],
              {
                "color": "rgba(115, 115, 115, 1)",
              },
            ],
          ]
        }
        testID="foo_locked_value_amount"
      >
        ≈ $123.00
      </Text>
    </View>
  </View>,
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexBasis": "0%",
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          [
            [
              {
                "fontSize": 12,
                "lineHeight": 16,
                "paddingRight": 56,
              },
              {
                "fontFamily": "MediumFont",
                "fontWeight": "500",
              },
            ],
            {
              "color": "rgba(115, 115, 115, 1)",
            },
          ],
        ]
      }
      testID="foo_available_label"
    >
      Available
    </Text>
    <View
      style={
        {
          "flexBasis": "0%",
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            [
              [
                {
                  "fontSize": 12,
                  "lineHeight": 16,
                },
                undefined,
              ],
              {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
          ]
        }
        testID="foo_available_amount"
      >
        99.00000000
      </Text>
    </View>
  </View>,
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexBasis": "0%",
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          [
            [
              {
                "fontSize": 12,
                "lineHeight": 16,
                "paddingRight": 56,
              },
              undefined,
            ],
            {
              "color": "rgba(115, 115, 115, 1)",
            },
          ],
        ]
      }
      testID="foo_available_value_label"
    />
    <View
      style={
        {
          "flexBasis": "0%",
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            [
              [
                {
                  "fontSize": 12,
                  "lineHeight": 16,
                },
                undefined,
              ],
              {
                "color": "rgba(115, 115, 115, 1)",
              },
            ],
          ]
        }
        testID="foo_available_value_amount"
      >
        ≈ $999,999.12
      </Text>
    </View>
  </View>,
]
`;
